source/common/router/header_parser.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <list>
+#include <string>
+
+#include "envoy/access_log/access_log.h"
+#include "envoy/http/header_map.h"
+
+#include "common/protobuf/protobuf.h"
+#include "common/router/header_formatter.h"
+
+#include "api/base.pb.h"
+
+namespace Envoy {
+namespace Router {
+
+class RequestHeaderParser;
+typedef std::unique_ptr<RequestHeaderParser> RequestHeaderParserPtr;
+
+/**
+ * This class provides request-time generation of upstream request headers. Header configurations
+ * are pre-parsed to select between constant values and values based on the evaluation of
+ * AccessLog::RequestInfo fields.
+ */
+class RequestHeaderParser {
+public:
+  virtual ~RequestHeaderParser() {}
+
+  static RequestHeaderParserPtr
+  parse(const Protobuf::RepeatedPtrField<envoy::api::v2::HeaderValueOption>& headers);
```

Comment:
I think I a name like parseConfig() or configure() or something would be clearer.

Reply:
I'd like to leave these unchanged for now -- they're not new, just moved files.